app/models/smart_proxy.rb

```diff
@@ -89,15 +89,15 @@ def has_feature?(feature_name)
   end
 
   def capabilities(feature)
-    smart_proxy_features.find_by(:feature_id => Feature.find_by(:name => feature)).try(:capabilities)
+    smart_proxy_features.find { |spf| spf.feature_name == feature }.try(:capabilities)
```

Comment:
Would it make sense to use `feature.to_s` to ensure it's always strings?

Reply:
what condition would it not be a string and we would want equality ?

Reply:
It may be a symbol, even though it would be wrong.

One thing I don't like in the Foreman codebase is that feature names are often strings all over the place. In the Katello codebase I think it's done better by using constants. If you use proper constants, there's little chance to make the mistake.

So overall, it's maybe more a question for my own understanding and less of a concrete suggestion.

Reply:
yes, the test failures actually brought this as an issue :)  updated to use .to_s which should cause them to pass